src/main/java/hng_java_boilerplate/notificationSettings/controller/NotificationSettingsController.java

```diff
@@ -0,0 +1,42 @@
+package hng_java_boilerplate.notificationSettings.controller;
+
+import hng_java_boilerplate.notificationSettings.entity.NotificationSettings;
+import hng_java_boilerplate.notificationSettings.service.NotificationSettingsService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RestController
+@RequestMapping
+public class NotificationSettingsController {
+    @Autowired
```

Comment:
Use constructor injection

Reply:
done

Reply:
I can't see constructor injection. Use @RequiredArgsConstructor and make the field final

Reply:
updated with @requiredargsconstructor